### PR TITLE
Add dataset - SemEval 2014 - Task 1

### PR DESCRIPTION
**Pull request by ashmeet13**
Adding the dataset of SemEval 2014 Task 1

Found the dataset under the shared Google Sheet > Recurring Task Datasets
Task Homepage - https://alt.qcri.org/semeval2014/task1

Thank you!

### CODE REVIEW

File: datasets/sem_eval_2014_task_1/sem_eval_2014_task_1.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    def _info(self):
+        return datasets.DatasetInfo(
+            # This is the description that will appear on the datasets page.
+            description=_DESCRIPTION,
+            # This defines the different columns of the dataset and their types
+            features=datasets.Features(
+                {
+                    "sentence_pair_id": datasets.Value("int64"),
+                    "sentence_A": datasets.Value("string"),
+                    "sentence_B": datasets.Value("string"),
```
Comment (lhoestq) [inline datasets/sem_eval_2014_task_1/sem_eval_2014_task_1.py:68]: could you rename them `premise` and `hypothesis` ? that's the standard we use for NLI datasets

**Review by lhoestq — COMMENTED** [review]
Looking pretty good so far !

Don't forget to add the tags. It's the yaml part at the top of the dataset card.
More info here : https://github.com/huggingface/datasets/blob/master/ADD_NEW_DATASET.md#tag-the-dataset-and-write-the-dataset-card

I also left a comment about the features names

File: datasets/sem_eval_2014_task_1/sem_eval_2014_task_1.py (hunk 1 above)
Comment (ashmeet13) [reply datasets/sem_eval_2014_task_1/sem_eval_2014_task_1.py]: Fixed this. Didn't know about the standard. My bad!

**Comment by ashmeet13** [comment]
Added the dataset card.
Requesting another review.

**Pull request merged**